src/main/java/com/microsoft/aad/adal4j/WSTrustRequest.java

```diff
@@ -58,37 +59,118 @@
                 MAX_EXPECTED_MESSAGE_SIZE);
-        messageBuilder.append(String.format(WSTRUST_ENVELOPE_TEMPLATE, guid,
-                resource, securityHeaderBuilder, DEFAULT_APPLIES_TO));
+        
+        String schemaLocation = "http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd";
+        String soapAction = "http://docs.oasis-open.org/ws-sx/ws-trust/200512/RST/Issue";
+        String rstTrustNamespace = "http://docs.oasis-open.org/ws-sx/ws-trust/200512";
+        String keyType = "http://docs.oasis-open.org/ws-sx/ws-trust/200512/Bearer";
+        String requestType = "http://docs.oasis-open.org/ws-sx/ws-trust/200512/Issue";
+        
+        if (addressVersion == WsTrustVersion.WSTRUST2005)
+        {
+            schemaLocation = "http://schemas.xmlsoap.org/ws/2005/02/trust/ws-trust.xsd";
+            soapAction = "http://schemas.xmlsoap.org/ws/2005/02/trust/RST/Issue";
+            rstTrustNamespace = "http://schemas.xmlsoap.org/ws/2005/02/trust";
+            keyType = "http://schemas.xmlsoap.org/ws/2005/05/identity/NoProofKey";
+            requestType = "http://schemas.xmlsoap.org/ws/2005/02/trust/Issue";
+        }
+
+
+        // Example WSTrust 1.3 request
+        // <s:Envelope xmlns:wst='http://schemas.xmlsoap.org/ws/2005/02/trust' xmlns:wssc='http://schemas.xmlsoap.org/ws/2005/02/sc' xmlns:wsa='http://www.w3.org/2005/08/addressing' xmlns:wsu='http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd' 
+        //          xmlns:wsp='http://schemas.xmlsoap.org/ws/2004/09/policy' xmlns:saml='urn:oasis:names:tc:SAML:1.0:assertion' xmlns:wsse='http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd' xmlns:ps='http://schemas.microsoft.com/Passport/SoapServices/PPCRL'
+        //          mlns:s='http://www.w3.org/2003/05/soap-envelope'>
+        //  <s:Header>
+        //      <wsa:Action s:mustUnderstand='1'>http://schemas.xmlsoap.org/ws/2005/02/trust/RST/Issue</wsa:Action>
+        //      <wsa:To s:mustUnderstand='1'>https://corp.sts.microsoft.com:443/adfs/services/trust/2005/windowstransport</wsa:To>
+        //      <wsa:MessageID>1303795308</wsa:MessageID>-<wsse:Security>-<wsu:Timestamp Id="Timestamp"><wsu:Created>2011-04-26T05:21:50Z</wsu:Created><wsu:Expires>2011-04-26T05:26:50Z</wsu:Expires></wsu:Timestamp></wsse:Security></s:Header>-<s:Body>-<wst:RequestSecurityToken Id="RST0"><wst:RequestType>http://schemas.xmlsoap.org/ws/2005/02/trust/Issue</wst:RequestType>-<wsp:AppliesTo>-<wsa:EndpointReference><wsa:Address>urn:federation:MicrosoftOnline</wsa:Address></wsa:EndpointReference></wsp:AppliesTo><wst:KeyType>http://schemas.xmlsoap.org/ws/2005/05/identity/NoProofKey</wst:KeyType></wst:RequestSecurityToken></s:Body></s:Envelope>
+        messageBuilder
+                .append(String
+                        .format("<s:Envelope xmlns:s='http://www.w3.org/2003/05/soap-envelope' xmlns:a='http://www.w3.org/2005/08/addressing' xmlns:u='%s'>"
```

Comment:
From http://specs.xmlsoap.org/ws/2005/02/trust/WS-Trust.pdf,
xmlns:a should be http://schemas.xmlsoap.org/ws/2004/08/addressing